src/main/webapp/scripts/sessioncache.js

```diff
@@ -33,60 +36,49 @@ class SessionCache {
      */
     this.sessionInformation_ = null;
 
-    /**
-     * @private {number}
-     */
-    this.refreshCadence_ = refreshCadence;
-
     /**
      * @private {Object}
      */
     this.urlParams_ = urlParams;
   }
 
-  /**
-   * Refreshes the result from the session information poller 
-   * and updates the sessionInformation field.
-   * Keys are updated every 30 seconds.
-   * @private
-   */
-  refreshSessionInformation_() {
-    throw new Error('Unimplemented');
-  }
-
   /** 
-   * This method begins polling for session information and starts
-   * refreshing.
+   * This method begins polling for session information.
    */
   start() {
-    throw new Error('Unimplemented');
+    this.sessionInformationPoller_.start();
   }
 
   /** 
-   * This method stops polling for session information and stops
-   * refreshing.
+   * This method stops polling for session information.
    */
   stop() {
-    throw new Error('Unimplemented');
+    this.sessionInformationPoller_.stop();
   }
 
   /**
    * Method sessionInformationRequest_() is the fetch api request
-   * responsible for gathering information about the current session 
-   * the client is in.
+   * responsible for gathering information about the current session.
    * @private
    */
-  sessionInformationRequest_() {
-    throw new Error('Unimplemented');
+  async sessionInformationRequest_() {
+    const /** string */ name = encodeURI(this.urlParams_.get('name'));
+    const /** string */ sessionID = 
+        encodeURI(this.urlParams_.get('session-id'));
+    const /** Object */ response = await fetch(
+        `/get-session-info?name=${name}&session-id=${sessionID}`);
+    return await response.json();
   }
```

Comment:
Disclaimer: I don't know javascript but:

Conceptually, I don't think this should be a member of the class. I think this should not be called anywhere outside of the Poller. In C++ I would suggest you declare/build this in the constructor and pass full ownership to the Poller, but I'm not sure what the equivalent is in JS. Maybe you define this within the constructor's scope so that it cannot be accessed outside of that?

@chukwuemekagooglecorp what are your thoughts here?

Reply:
I think local functions in the constructor are possible!

Reply:
JS doesnt do class scoping so everything is public. The underscore at the end is convention for private so no one should be calling this outside of the class. But, why give them the opportunity to? I think Danny is right. Build this function in the cache constructor and pass it to the poller there